assets/src/block-validation/add-toolbar-button-to-block.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect, withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { BLOCK_VALIDATION_STORE_KEY } from './store';
+import { AMPToolbarButton } from './amp-toolbar-button';
+
+/**
+ * Filters blocks edit function of all blocks.
+ *
+ * @param {Function} BlockEdit function.
+ *
+ * @return {Function} Edit function.
+ */
+export function addToolbarButtonToBlock( BlockEdit ) {
+	// Filter a class component.
+	if ( BlockEdit.prototype.render ) {
+		return withSelect(
+			( select, ownProps ) => ( {
+				count: select( BLOCK_VALIDATION_STORE_KEY ).getBlockValidationErrors( ownProps.clientId )?.length || 0,
+			} ) )(
+			class BlockEditWithAmpToolBarButtonClass extends BlockEdit {
+				render() {
+					return (
+						<>
+							{ 0 < this.props.count && (
+								<AMPToolbarButton count={ this.props.count } />
+							) }
+							{ super.render() }
+						</>
+					);
+				}
+			} );
+	}
```

Comment:
I don't think this is necessary at all..?

The below version should suffice, just rendering the `<BlockEdit>` component as-is + the toolbar button.

https://developer.wordpress.org/block-editor/developers/filters/block-filters/#editor-blockedit

Reply:
Yup, that worked: 63ae231b4 I was making it too complicated.